pkg/scheduler/framework/plugins/noderesources/fit.go

```diff
@@ -244,11 +247,118 @@ func (f *Fit) EventsToRegister() []framework.ClusterEventWithHint {
 		podActionType |= framework.Update
 	}
 	return []framework.ClusterEventWithHint{
-		{Event: framework.ClusterEvent{Resource: framework.Pod, ActionType: podActionType}},
+		{Event: framework.ClusterEvent{Resource: framework.Pod, ActionType: podActionType}, QueueingHintFn: f.isSchedulableAfterPodChange},
+		// TODO: Ideally, it's supposed to add a QueueingHintFn for verifying whether the Pod becomes schedulable after receiving
+		// events of NodeCreated and NodeUpdated. But, we may miss NodeCreated event due to preCheck. It may cause the Pod to be
+		// stuck in the scheduling queue forever.
+		// We have to wait for implementing QueueingHint until preCheck is completely removed.
+		// See: https://github.com/kubernetes/kubernetes/issues/110175
 		{Event: framework.ClusterEvent{Resource: framework.Node, ActionType: framework.Add | framework.Update}},
 	}
 }
 
+// isSchedulableAfterPodChange is invoked whenever a pod deleted or updated. It checks whether
+// that change made a previously unschedulable pod schedulable.
+func (f *Fit) isSchedulableAfterPodChange(logger klog.Logger, pod *v1.Pod, oldObj, newObj interface{}) (framework.QueueingHint, error) {
+	originalPod, modifiedPod, err := schedutil.As[*v1.Pod](oldObj, newObj)
+	if err != nil {
+		return framework.Queue, err
+	}
+
+	if modifiedPod == nil {
+		logger.V(4).Info("an other pod was deleted", "pod", klog.KObj(pod), "otherPod", klog.KObj(originalPod))
```

Comment:
Also, can you change the log level to 5. (other logs as well

> If you increases the log level to more than 5, you can see the logs related to QueueingHint in the scheduling queue. Also, the in-tree plugins emits all logs in QueueingHint with log level 5. (If you have a custom plugin, you may want to check the log level in its QueueingHint.)
https://github.com/kubernetes/enhancements/blob/master/keps/sig-scheduling/4247-queueinghint/README.md#what-are-other-known-failure-modes